src/main/scala/nl/knaw/dans/lib/logging/DebugEnhancedLogging.scala

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (C) 2016 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.lib.logging
+
+import com.typesafe.scalalogging.LazyLogging
+
+trait DebugEnhancedLogging extends LazyLogging {
+  /**
+   * Logs a trace level statement with the function name, parameter names and parameter values.
+   *
+   * Example:
+   * {{{
+   *   // Function definition
+   *   def someFunction(s: String, i: Int) = {
+   *    trace(s, i)
+   *    // Your code ...
+   *   }
+   *
+   *   // Function call
+   *   someFunction("My input string", 2)
+   *
+   *   // This will result in a logging line like this (if the TRACE log level is enabled):
+   *   [TRACE] someFunction [(s, i)]:  ("My input string",2)
+   *
+   * }}}
+   *
+   * @param value list of parameter values
+   * @param enclosing implicitly passed in context from the sourcecode library
+   * @tparam V type parameter, no idea how it works
+   */
+  def trace[V](value: sourcecode.Text[V])(implicit enclosing: sourcecode.Name) = {
+    logger.trace(s"${enclosing.value} [${value.source}]: ${value.value}")
+  }
+
+
+  /**
+   * Logs a custom debug level statement with the calling function name prefixed
+   *
+   * Example:
+   * {{{
+   *   // Function definition
+   *   def someFunction(s: String, i: Int) = {
+   *    // Your code ...
+   *    debug("Doing important stuff here")
+   *   }
+   *
+   *
+   *   // This will result in a logging line like this (if the DEBUG log level is enabled):
+   *   [DEBUG] someFunction: Doing important stuff here
+   *
+   * }}}
+   * @param s
+   * @param enclosing
+   */
+  def debug(s: String)(implicit enclosing: sourcecode.Name) = logger.debug(s"${enclosing.value}: $s")
```

Comment:
Note that logback.xml can be configured to even produce the line number. When within your IDE you can click that number to jump to the code. At least that is what I remember from Java and Eclipse... http://logback.qos.ch/manual/configuration.html

    <appender name="FILE" class="ch.qos.logback.core.FileAppender">
      <file>myApp.log</file>
      <encoder>
        <pattern>%date %level [%thread] %logger{10} [%file:%line] %msg%n</pattern>
      </encoder>
    </appender>

Having the method name might be clearer at first sight in some situations when not troubled with too much $anon. Otherwise I don't see very much added value. A full stack trace with just method names on a single line might be more helpful.

We are also used to log with `logger.something` I'd rather have another something otherwise I forget about these trace and debug. But maybe I'm missing some clue.